src/EnergyPlus/DXCoils.cc

```diff
@@ -2979,895 +2979,978 @@ void GetDXCoils(EnergyPlusData &state)
-            thisDXCoil.PLFFPLR(1) = GetCurveIndex(state, Alphas(17));
-            if (thisDXCoil.PLFFPLR(1) == 0) {
-                ShowSevereError(state, format("{}{}=\"{}\", invalid", RoutineName, CurrentModuleObject, thisDXCoil.Name));
-                ShowContinueError(state, format("...not found {}=\"{}\".", cAlphaFields(17), Alphas(17)));
-                ErrorsFound = true;
-            } else {
-                // Verify Curve Object, only legal types are Cubic or Quadratic
-                ErrorsFound |= Curve::CheckCurveDims(state,
-                                                     thisDXCoil.PLFFPLR(1), // Curve index
-                                                     {1},                   // Valid dimensions
-                                                     RoutineName,           // Routine name
-                                                     CurrentModuleObject,   // Object Type
-                                                     thisDXCoil.Name,       // Object Name
-                                                     cAlphaFields(17));     // Field Name
 
-                if (!ErrorsFound) {
-                    //       Test PLF curve minimum and maximum. Cap if less than 0.7 or greater than 1.0.
-                    MinCurveVal = 999.0;
-                    MaxCurveVal = -999.0;
-                    CurveInput = 0.0;
-                    while (CurveInput <= 1.0) {
-                        CurveVal = CurveValue(state, thisDXCoil.PLFFPLR(1), CurveInput);
-                        if (CurveVal < MinCurveVal) {
-                            MinCurveVal = CurveVal;
-                            MinCurvePLR = CurveInput;
-                        }
-                        if (CurveVal > MaxCurveVal) {
-                            MaxCurveVal = CurveVal;
-                            MaxCurvePLR = CurveInput;
-                        }
-                        CurveInput += 0.01;
-                    }
-                    if (MinCurveVal < 0.7) {
-                        ShowWarningError(state, format("{}{}=\"{}\", invalid", RoutineName, CurrentModuleObject, thisDXCoil.Name));
-                        ShowContinueError(state, format("...{} = {} has out of range value.", cAlphaFields(17), Alphas(17)));
-                        ShowContinueError(state,
-                                          format("...Curve minimum must be >= 0.7, curve min at PLR = {:.2T} is {:.3T}", MinCurvePLR, MinCurveVal));
-                        ShowContinueError(state, "...Setting curve minimum to 0.7 and simulation continues.");
-                        Curve::SetCurveOutputMinValue(state, thisDXCoil.PLFFPLR(1), ErrorsFound, 0.7);
-                    }
+            cFieldName = "Heating COP Function of Water Flow Fraction Curve Name"; // Alphas(16)
+            fieldValue = s_ip->getAlphaFieldValue(fields, schemaProps, "heating_cop_function_of_water_flow_fraction_curve_name");
+            if (!fieldValue.empty()) {
```

Comment:
Something else which is neither here nor there but is just good programming IMO is that it's better to "linearize" this kind of logic rather than nest it, e.g., 

```cpp
if (fieldValue.empty()) { 
   // This empty body is the same as nesting on if (!fieldValue.empty())
} else if ((thisDXcoil.HCOPWaterFlow = GetCurveIndex(state, fieldValue)) == 0) { // assigment and comparison one liner
  ShowSevereError();
} else if (CheckCurveDims(state, ...)) {
  ShowSevereError();
} else {
   // Process curve data
}
```

Reply:
I modified one instance of such an occurrence in the Water2Air Heat Pump. Some of the error message functions do not match the existing error message exactly. I did my best to capture the message using the functions. Please review and provide comments before I propagate similar changes to other DX coils. @amirroth @Myoldmopar.